.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) Open Enclave SDK contributors.
+# Licensed under the MIT License.
+
+name: "CodeQL"
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ master ]
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'cpp' ]
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      with:
+        languages: ${{ matrix.language }}
+        config-file: openenclave/openenclave-security/src/static/codeql/queries/cpp/suites/oe-codeql-security-queries.qls@main
+
+    - name: Install Dependencies
+      run: |
+        sudo apt install -y doxygen
+        git submodule update --init --recursive
+
+    - name: Build
+      run: |
+       mkdir build
+       cd build
+       cmake .. -DBUILD_TESTS=OFF
```

Comment:
DBUILD_TESTS is not required for oeedger8r project.
just "cmake .." would do.

or CodeQL default "autobuild"
    - name: Autobuild
      uses: github/codeql-action/autobuild@v1